service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -361,4 +387,33 @@ public PageResponse<List<PipelineRun>> findPipelineRunsPaginated(
   public long getPipelineRunCount(String userId) {
     return pipelineRunsRepository.countByUserId(userId);
   }
+
+  private Sort.Direction validateSortDirection(String sortDirection) {
+    if (sortDirection == null) {
+      return Sort.Direction.DESC;
+    }
+    try {
+      return Sort.Direction.valueOf(sortDirection.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(
+          "Invalid sort direction: %s. Valid values are ASC or DESC".formatted(sortDirection));
+    }
+  }
+
+  private String validateSortProperty(String sortProperty) {
+    // default to sorting by created
+    if (sortProperty == null) {
+      return "created";
+    }
+
+    List<String> allowedSortProperties = List.of("created", "updated", "quotaConsumed");
```

Comment:
is there any way to verify that these fields exist in the underlying entity being paginated (PipelineRun)?  Might be nice to just catch ourselves if we change the db tables but forget to update this allowed list.  Could be here or possibly a test.